### ISSUE TITLE
'NoneType' object has no attribute 'objects'

### ISSUE THREAD

**Issue by xmeltrut** [wagtail/wagtail#3751]
When I move a model from one app's `models.py` to another, I then get the following error:

    AttributeError at /admin/pages/5/
    'NoneType' object has no attribute 'objects'

My guess is that it is happening because the model has disappeared, but is still referenced in the database. So, in that respect, it is my fault. However, I think such errors should be handled more gracefully as this seems a plausible scenario that people will run into. Currently, it brings down the whole admin.

**Comment by gasman** [comment]
Hi @xmeltrut - please can you give the full stack trace of the error, and the versions of Wagtail / Django / Python you're using?

**Comment by xmeltrut** [comment]
I'm using:

Python: `3.6.2`
Django: `1.11.4`
Wagtail: `1.11.1`

Here is the stack trace:

```
Environment:


Request Method: GET
Request URL: http://localhost:8000/admin/pages/4/

Django Version: 1.11.4
Python Version: 3.6.2
Installed Applications:
['home',
 'search',
 'blog',
 'wagtail.wagtailforms',
 'wagtail.wagtailredirects',
 'wagtail.wagtailembeds',
 'wagtail.wagtailsites',
 'wagtail.wagtailusers',
 'wagtail.wagtailsnippets',
 'wagtail.wagtaildocs',
 'wagtail.wagtailimages',
 'wagtail.wagtailsearch',
 'wagtail.wagtailadmin',
 'wagtail.wagtailcore',
 'modelcluster',
 'taggit',
 'django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles']
Installed Middleware:
['django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.auth.middleware.SessionAuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'django.middleware.security.SecurityMiddleware',
 'wagtail.wagtailcore.middleware.SiteMiddleware',
 'wagtail.wagtailredirects.middleware.RedirectMiddleware']


Template error:
In template C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\wagtail\wagtailadmin\templates\wagtailadmin\pages\listing\_list.html, error at line 46
   'NoneType' object has no attribute 'objects'   36 :                     {% endif %}
   37 :                 </td>
   38 :                 <td></td>
   39 :             </tr>
   40 :         {% endif %}
   41 : 
   42 :         {% block post_parent_page_headers %}
   43 :         {% endblock %}
   44 :     </thead>
   45 :     <tbody>
   46 :          {% if pages %} 
   47 :             {% for page in pages %}
   48 :                 {% page_permissions page as page_perms %}
   49 :                 <tr {% if ordering == "ord" %}id="page_{{ page.id|unlocalize }}" data-page-title="{{ page.get_admin_display_title }}"{% endif %} class="{% if not page.live %}unpublished{% endif %} {% block page_row_classname %}{% endblock %}">
   50 :                     {% if show_ordering_column %}
   51 :                         <td class="ord">{% if orderable and ordering == "ord" %}<div class="handle icon icon-grip text-replace">{% trans 'Drag' %}</div>{% endif %}</td>
   52 :                     {% endif %}
   53 :                     <td class="title" valign="top" data-listing-page-title>
   54 :                         {% block page_title %}
   55 :                         {% endblock %}
   56 :                     </td>


Traceback:

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\core\handlers\exception.py" in inner
  41.             response = get_response(request)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\core\handlers\base.py" in _get_response
  187.                 response = self.process_exception_by_middleware(e, request)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\core\handlers\base.py" in _get_response
  185.                 response = wrapped_callback(request, *callback_args, **callback_kwargs)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\views\decorators\cache.py" in _cache_controlled
  43.             response = viewfunc(request, *args, **kw)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\wagtail\wagtailadmin\decorators.py" in decorated_view
  31.             return view_func(request, *args, **kwargs)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\wagtail\wagtailadmin\utils.py" in wrapped_view_func
  139.                 return view_func(request, *args, **kwargs)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\wagtail\wagtailadmin\views\pages.py" in index
  102.         'do_paginate': do_paginate,

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\shortcuts.py" in render
  30.     content = loader.render_to_string(template_name, context, request, using=using)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\template\loader.py" in render_to_string
  68.     return template.render(context, request)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\template\backends\django.py" in render
  66.             return self.template.render(context)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\template\base.py" in render
  207.                     return self._render(context)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\template\base.py" in _render
  199.         return self.nodelist.render(context)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\template\base.py" in render
  990.                 bit = node.render_annotated(context)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\template\base.py" in render_annotated
  957.             return self.render(context)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\template\loader_tags.py" in render
  177.             return compiled_parent._render(context)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\template\base.py" in _render
  199.         return self.nodelist.render(context)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\template\base.py" in render
  990.                 bit = node.render_annotated(context)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\template\base.py" in render_annotated
  957.             return self.render(context)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\template\loader_tags.py" in render
  177.             return compiled_parent._render(context)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\template\base.py" in _render
  199.         return self.nodelist.render(context)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\template\base.py" in render
  990.                 bit = node.render_annotated(context)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\template\base.py" in render_annotated
  957.             return self.render(context)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\template\loader_tags.py" in render
  177.             return compiled_parent._render(context)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\template\base.py" in _render
  199.         return self.nodelist.render(context)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\template\base.py" in render
  990.                 bit = node.render_annotated(context)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\template\base.py" in render_annotated
  957.             return self.render(context)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\template\loader_tags.py" in render
  72.                 result = block.nodelist.render(context)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\template\base.py" in render
  990.                 bit = node.render_annotated(context)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\template\base.py" in render_annotated
  957.             return self.render(context)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\template\loader_tags.py" in render
  72.                 result = block.nodelist.render(context)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\template\base.py" in render
  990.                 bit = node.render_annotated(context)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\template\base.py" in render_annotated
  957.             return self.render(context)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\template\loader_tags.py" in render
  216.                 return template.render(context)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\template\base.py" in render
  209.                 return self._render(context)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\template\base.py" in _render
  199.         return self.nodelist.render(context)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\template\base.py" in render
  990.                 bit = node.render_annotated(context)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\template\base.py" in render_annotated
  957.             return self.render(context)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\template\loader_tags.py" in render
  177.             return compiled_parent._render(context)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\template\base.py" in _render
  199.         return self.nodelist.render(context)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\template\base.py" in render
  990.                 bit = node.render_annotated(context)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\template\base.py" in render_annotated
  957.             return self.render(context)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\template\defaulttags.py" in render
  321.             if match:

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\core\paginator.py" in __len__
  137.         return len(self.object_list)

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\db\models\query.py" in __len__
  232.         self._fetch_all()

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\django\db\models\query.py" in _fetch_all
  1118.             self._result_cache = list(self._iterable_class(self))

File "C:\Users\chris\AppData\Local\Programs\Python\Python36-32\lib\site-packages\wagtail\wagtailcore\query.py" in specific_iterator
  380.         pages = model.objects.filter(pk__in=pks)

Exception Type: AttributeError at /admin/pages/4/
Exception Value: 'NoneType' object has no attribute 'objects'

```

**Comment by jorenham** [comment]
I'm facing the exact same issue

**Comment by jorenham** [comment]
I found a workaround!
Delete the `django.contrib.contenttypes.models.ContentType` object associated with the deleted model.
So I've you've removed `yourapp.models.YourModel` you do 

```python
from django.contrib.contenttypes.models import ContentType

ContentType.objects.get(label='yourapp', model='yourmodel').delete()
```

**Comment by gasman** [comment]
This is fixed in the forthcoming 1.12, by #3569.
